### ISSUE TITLE
mBART50 finetuned N-N model failed to generate translation

### ISSUE THREAD

**Issue by Mao-KU** [facebookresearch/fairseq#3932]
Hello, I downloaded mBART50 finetuned many-to-many model. When generating translations, it just repeats the same word over and over again. 

This problem is also reported in https://github.com/huggingface/transformers/issues/7060#issuecomment-765648270.

I used mBART50 finetuned many-to-many model to do the ar_AR-en_XX translation on the IWSLT17 ar-en test set.
No bug occurred during the generation.
Also, I have done the multilingual fine-tuning for mBART25 successfully, so I suppose I did the preprocessing correctly.

The file that mBART50 fine-tuned N-N generated is put here:
https://lotus.kuee.kyoto-u.ac.jp/~zhuoyuanmao/ar_AR-en_XX

Command for generating translation:
lang_pairs="en_XX-ja_XX,ja_XX-en_XX,it_IT-en_XX,en_XX-it_IT,en_XX-nl_XX,nl_XX-en_XX,en_XX-ar_AR,ar_AR-en_XX,en_XX-vi_VN,vi_VN-en_XX"
path_2_data=data-bin/mbart_626_en-it-vi-ja-nl-ar
lang_list=ML50_langs.txt # <path to a file which contains list of languages separted by new lines>
model_path=mbart50.ft.nn/model.pt 
source_lang=ar_AR
target_lang=en_XX
export CUDA_DEVICE_ORDER=PCI_BUS_ID
export CUDA_VISIBLE_DEVICES=1
fairseq-generate ${path_2_data} \
  --path ${model_path} \
  --task translation_multi_simple_epoch \
  --gen-subset test \
  --source-lang ${source_lang} \
  --target-lang ${target_lang} \
  --sacrebleu --remove-bpe 'sentencepiece'\
  --batch-size 128 \
  --encoder-langtok "src" \
  --decoder-langtok \
  --lang-dict "$lang_list" \
  --lang-pairs "$lang_pairs" > ${model_path}/${source_lang}-${target_lang}

 - fairseq Version (e.g., 1.0 or main): 1.0
 - PyTorch Version (e.g., 1.0): 1.9.1
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source): pip install --editable ./
 - Python version: 3.8
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: V100 32G
 - Any other relevant information:
